src/wp-includes/interactivity-api/class-wp-interactivity-api.php

```diff
@@ -0,0 +1,930 @@
+		// In closing tags, it removes the last context from the stack.
+		if ( $p->is_tag_closer() ) {
+			array_pop( $context_stack );
+			return;
+		}
+
+		$attribute_value = $p->get_attribute( 'data-wp-context' );
+		$namespace_value = end( $namespace_stack );
+
+		// Separates the namespace from the context JSON object.
+		list( $namespace_value, $decoded_json ) = is_string( $attribute_value ) && ! empty( $attribute_value )
+			? $this->extract_directive_value( $attribute_value, $namespace_value )
+			: array( $namespace_value, null );
+
+		/*
+		 * If there is a namespace, it adds a new context to the stack merging the
+		 * previous context with the new one.
+		 */
+		if ( is_string( $namespace_value ) ) {
+			$context_stack[] = array_replace_recursive(
+				end( $context_stack ) !== false ? end( $context_stack ) : array(),
+				array( $namespace_value => is_array( $decoded_json ) ? $decoded_json : array() )
+			);
+		} else {
+			/*
+			 * If there is no namespace, it pushes the current context to the stack.
+			 * It needs to do so because the function pops out the current context
+			 * from the stack whenever it finds a `data-wp-context`'s closing tag.
+			 */
+			$context_stack[] = end( $context_stack );
+		}
+	}
+
+	/**
+	 * Processes the `data-wp-bind` directive.
+	 *
+	 * It updates or removes the bound attributes based on the evaluation of its
+	 * associated reference.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param WP_Interactivity_API_Directives_Processor $p               The directives processor instance.
+	 * @param array                                     $context_stack   The reference to the context stack.
+	 * @param array                                     $namespace_stack The reference to the store namespace stack.
+	 */
+	private function data_wp_bind_processor( WP_Interactivity_API_Directives_Processor $p, array &$context_stack, array &$namespace_stack ) {
+		if ( ! $p->is_tag_closer() ) {
+			$all_bind_directives = $p->get_attribute_names_with_prefix( 'data-wp-bind--' );
+
+			foreach ( $all_bind_directives as $attribute_name ) {
+				list( , $bound_attribute ) = $this->extract_prefix_and_suffix( $attribute_name );
+				if ( empty( $bound_attribute ) ) {
+					return;
+				}
+
+				$attribute_value = $p->get_attribute( $attribute_name );
+				$result          = $this->evaluate( $attribute_value, end( $namespace_stack ), end( $context_stack ) );
+
+				if ( null !== $result && ( false !== $result || '-' === $bound_attribute[4] ) ) {
+					/*
+					 * If the result of the evaluation is a boolean and the attribute is
+					 * `aria-` or `data-, convert it to a string "true" or "false". It
+					 * follows the exact same logic as Preact because it needs to
+					 * replicate what Preact will later do in the client:
+					 * https://github.com/preactjs/preact/blob/ea49f7a0f9d1ff2c98c0bdd66aa0cbc583055246/src/diff/props.js#L131C24-L136
+					 */
+					if ( is_bool( $result ) && '-' === $bound_attribute[4] ) {
```

Comment:
Looking for the 5th character to be a hyphen is clever, but perhaps it could accidentally match more attributes than just `data-` or `aria-`, for example `http-equiv` or any other number of custom HTML attributes. Seems safer to be explicit:

```suggestion
				$is_aria_or_data = str_starts_with( $bound_attribute, 'aria-' ) || str_starts_with( $bound_attribute, 'data-' );
				if ( null !== $result && ( false !== $result || '-' === $bound_attribute[4] ) ) {
					/*
					 * If the result of the evaluation is a boolean and the attribute is
					 * `aria-` or `data-, convert it to a string "true" or "false". It
					 * follows the exact same logic as Preact because it needs to
					 * replicate what Preact will later do in the client:
					 * https://github.com/preactjs/preact/blob/ea49f7a0f9d1ff2c98c0bdd66aa0cbc583055246/src/diff/props.js#L131C24-L136
					 */
					if ( is_bool( $result ) && $is_aria_or_data ) {
```

Reply:
I wouldn't mind checking directly for `data-` or `aria-`, but as it's explained in the comment, we need to follow what Preact will do in the client, otherwise the result of the server directive processing could be different than the result of the interactivity in the client, and that would be confusing.

> It follows the exact same logic as Preact because it needs to replicate what Preact will later do in the client: https://github.com/preactjs/preact/blob/ea49f7a0f9d1ff2c98c0bdd66aa0cbc583055246/src/diff/props.js#L131C24-L136

We can open an issue in Preact and ask them to change this if you think this is important for custom attributes. 

For valid HTML, other than `http-equiv` (which should not be a problem because it requires a string value), I can't find any other HTML attribute that will conflict.

Reply:
Ah, I see. OK. Makes sense then to keep as-is.

Reply:
Ok, resolving then 🙂